### PR TITLE
cgroup: Some memory protection fixes

### PR DESCRIPTION
**Pull request by cdown**
1. Add docs and NEWS entry for DefaultMemoryMin and friends
2. Fix up DefaultMemoryMin, somehow I didn't send upstream version with full plumbing
3. Support 0-values for memory protection -- this was always a bit wrong, but now it's even more important to fix with DefaultMemoryLow=N and a child having MemoryLow=0 being a totally legitimate configuration

### CODE REVIEW

**Comment by cdown** [comment]
(the travis ci issue looks nonsensical, so pushing again)

**Comment by cdown** [comment]
@poettering, @keszybz, just a poke about this one. This should be ready for review, amd64 failure is unrelated. We should definitely get this in before v243 to avoid having another case of 0 meaning different things between different versions.

File: NEWS [hunk 1]
```diff
@@ -19,6 +19,11 @@ CHANGES WITH 243 in spe:
           are harder to type, but we believe the change from 5 digit PIDs to 7
           digit PIDs is not too hampering for usability.
 
+        * MemoryLow and MemoryMin gained hierarchically-aware counterparts,
```
Comment (keszybz) [inline NEWS]: I think you mean "hierarchy-aware".

File: man/systemd.resource-control.xml [hunk 2]
```diff
@@ -245,6 +245,10 @@
 
           <para>This setting is supported only if the unified control group hierarchy is used and disables
           <varname>MemoryLimit=</varname>.</para>
+
+          <para>Units may can have their children use a default <literal>memory.min</literal> value by specifying
+          <varname>DefaultMemoryMin=</varname>, which has the same usage as <varname>MemoryMin=</varname>. This setting
```
Comment (keszybz) [inline man/systemd.resource-control.xml]: usage → semantics ?

File: man/systemd.resource-control.xml [hunk 3]
```diff
@@ -245,6 +245,10 @@
 
           <para>This setting is supported only if the unified control group hierarchy is used and disables
           <varname>MemoryLimit=</varname>.</para>
+
+          <para>Units may can have their children use a default <literal>memory.min</literal> value by specifying
```
Comment (keszybz) [inline man/systemd.resource-control.xml]: "may can"

**Comment by cdown** [comment]
Thanks, updated. PTAL.

**Comment by keszybz** [comment]
travis-ci:
`Assertion 'path_equal(p, t)' failed at ../src/test/test-mountpoint-util.c:93, function test_mnt_id(). Aborting.`
Hmm.

**Pull request merged**